packages/bp/src/core/dialog/dialog-engine.ts

```diff
@@ -211,6 +211,55 @@ export class DialogEngine {
     }
   }
 
+  private findNodeWithoutError = (botId, flow, nodeName) => {
+    try {
+      return this._findNode(botId, flow, nodeName)
+    } catch (err) {
+      // ignore
+    }
+    return undefined
+  }
+
+  private findFlowWithoutError = (botId, flowName) => {
+    try {
+      return this._findFlow(botId, flowName)
+    } catch (err) {
+      // ignore
+    }
+    return undefined
+  }
+
+  private fillContextForTransition = (
+    event: IO.IncomingEvent,
+    {
+      currentFlow,
+      currentNode,
+      nextFlow,
+      nextNode
+    }: {
+      currentFlow: FlowWithParent
+      currentNode?: FlowNode
+      nextFlow: FlowWithParent
+      nextNode: FlowNode
+    }
+  ) => {
+    if (event.state.context) {
+      event.state.context.currentNode = nextNode.name
+      event.state.context.currentFlow = nextFlow.name
+      event.state.context.queue = undefined
+      event.state.context.previousFlow = currentFlow.name
+      event.state.context.previousNode = currentNode?.name as string
+      event.state.context.jumpPoints = [
+        ...(event.state.context?.jumpPoints || []),
+        {
+          flow: currentFlow.name,
+          node: currentNode?.name as string
+        }
+      ]
+      event.state.context.hasJumped = true
+    }
```

Comment:
Can you invert this so there's less indentation?

Reply:
Let me know what you think of my new approach